components/skus/browser/rs/lib/src/sdk/credentials/mod.rs

```diff
@@ -182,14 +182,29 @@ where
         domain: &str,
     ) -> Result<Option<CredentialSummary>, SkusError> {
         if let Some(orders) = self.client.get_orders().await? {
-            for order in orders {
-                if order.location_matches(&self.environment, domain) {
-                    let wrapped_value =
-                        self.matching_order_credential_summary(&order.id, domain).await;
-                    if wrapped_value.is_err() {
-                        continue;
+            let mut orders: Vec<_> = orders
+                .into_iter()
+                .filter(|order| order.location_matches(&self.environment, domain))
+                .collect();
+            orders.sort_by(|a, b| b.expires_at.cmp(&a.expires_at));
+            if let Some(order) = orders.first() {
+                // We have at least one order for the specified location
+                if let Ok(Some(summary)) =
+                    self.matching_order_credential_summary(&order.id, domain).await
+                {
+                    return Ok(Some(summary));
+                } else {
+                    if let Some(expires_at) = order.expires_at {
```

Comment:
Thanks for adding a comment. In general this code would benefit from a few more. 

It looks like if there's a credential matching the domain and a summary can be created, it's returned. If not, and the first credential happens to be unexpired, a summary is created for it directly. What is the intent here?

`matching_order_credential_summary` refreshes the credentials for the order if they're expired. Why is that not necessary here? It also checks for domain match again, so why not just call it up front, before sorting by expiry time? Would that be a performance issue?

Reply:
The high level motivation for this change was to give callers of the CredentialSummary in the SKU SDK a way to differentiate between having an inactive subscription vs. having an active subscription with only expired credentials (requiring users to log in to account.brave.com to refresh).  Previously, CredentialSummary was returning `{}` for both of those case.  Now, the CredentialSummary returns a non empty object in the second case that has `remaining_credential_count: 0` as a key and value.

As far as the specifics of the code, I'm not as familiar. Maybe @evq can comment as they provided me this patch as well.

Reply:
Ok, if the `remaining_credential_count` is the important point, that makes more sense. Thanks for clarifying; feel free to document that in the actual code. :)

Reply:
cc @bsclifton since this will change the behavior of the SKU SDK, potentially impacting it's use for VPN.  Based on my reading of this [code](https://github.com/brave/brave-core/blob/f54931821e3dd5cddace52b9a363190a5038c9b5/components/brave_vpn/browser/brave_vpn_service_helper.cc#L44-L50), though it seems you are already expecting the SDK to behave in the way I've updated it